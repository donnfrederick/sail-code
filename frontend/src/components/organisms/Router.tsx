import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
