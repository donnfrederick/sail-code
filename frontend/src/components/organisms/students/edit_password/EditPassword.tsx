import Button from 'components/atoms/students/Button'
import PasswordInput from 'components/molecules/students/PasswordInput'
import * as StudentsModels from 'models/students'
import * as React from 'react'
import styled from 'styled-components'

interface Props {
  authToken: string
  error: any
  info: StudentsModels.Info
  patchMe(authToken: string, changes: any): void
  register(info: StudentsModels.Info): void
}

export default (props: Props) => {
  const { authToken, error, info, patchMe, register } = props

  const getChanges = () => ({
    current_password: info.current_password,
    password: info.password
  })

  return (
    <Container>
      <PasswordInput
        error={error}
        info={info}
        isCurrentPassword={true}
        label={'Current Password'}
        register={register}
      />
      <PasswordInput
        error={error}
        info={info}
        label={'New Password'}
        register={register}
      />
      <ButtonContainer>
        <Button
          type="blue"
          text="Save"
          isActive={
            info.current_password.length >= 8 && info.password.length >= 8
          }
          onClick={async () => {
            try {
              await patchMe(authToken, getChanges())
            } catch (error) {
              // tslint:disable-next-line
            console.error(error)
            }
          }}
        />
      </ButtonContainer>
    </Container>
  )
}

const Container = styled.div`
  width: 100%;
  box-sizing: border-box;
  padding: 0 64px;
  text-align: center;
`

const ButtonContainer = styled.div`
  display: flex;
  justify-content: center;
  position: absolute;
  right: 0;
  left: 0;
  bottom: 156px;
`
