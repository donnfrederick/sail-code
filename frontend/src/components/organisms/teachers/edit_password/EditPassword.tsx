import { Info } from 'models/teachers'
import * as React from 'react'
import styled from 'styled-components'
import * as validate from 'utils/validate'

interface Props {
  error: any
  info: Info
  register(info: Info): void
  setCurrentPassword(password: string): void
}

export default (props: Props) => {
  const { error, info, register, setCurrentPassword } = props

  const hasPasswordError = validate.hasError('パスワード')(error)
  const passwordErrorMessage = validate.getErrorMessage('パスワード')(error)

  return (
    <Container>
      <Label data-error={hasPasswordError ? true : false}>
        {passwordErrorMessage || '現在のパスワード'}
      </Label>
      <Input
        data-error={hasPasswordError}
        type="password"
        pattern="^[0-9A-Za-z]+$"
        maxLength={72}
        placeholder="半角英数8字以上"
        onInput={(event: any) => {
          const input = event.target.value
          setCurrentPassword(input)
        }}
      />
      <Label data-error={hasPasswordError ? true : false}>
        {passwordErrorMessage || '新しいパスワード'}
      </Label>
      <Input
        data-error={hasPasswordError}
        type="password"
        pattern="^[0-9A-Za-z]+$"
        maxLength={72}
        placeholder="半角英数8字以上"
        onInput={(event: any) => {
          const input = event.target.value
          info.password = input
          register(info)
        }}
      />
    </Container>
  )
}

const Container = styled.div`
  width: 640px;
  margin: 0 auto 56px;
`

const Label = styled.div`
  height: 32px;
  margin-bottom: 16px;
  font-size: 32px;
  letter-spacing: 0px;
  text-align: left;
  color: #405766;

  &[data-error='true'] {
    color: #fa6970;
  }
`

const Input = styled.input`
  appearance: none;
  width: 100%;
  height: 100px;
  opacity: 0.6;
  box-sizing: border-box;
  margin-bottom: 56px;
  padding: 0 32px;
  border: solid 3px transparent;
  border-radius: 8px;
  outline: none;
  background-color: #e3edf5;
  font-size: 36px;
  font-weight: 500;
  line-height: 100px;
  letter-spacing: 0px;
  text-align: left;
  color: #3e84b3;

  &:last-child {
    margin-bottom: 0;
  }

  &:focus {
    border: solid 3px #59b4ff;
  }

  &[data-error='true'] {
    border: solid 3px #fa6970;
    background-color: #f5e3e3;
    color: #d13e3e;
  }
`
