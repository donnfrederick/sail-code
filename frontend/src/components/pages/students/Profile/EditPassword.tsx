import StudentsProfileEditPassword from 'components/templates/students/profile/EditPassword'
import * as React from 'react'

export default () => {
  return <StudentsProfileEditPassword />
}
