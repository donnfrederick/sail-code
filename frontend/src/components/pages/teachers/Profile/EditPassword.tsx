import TeachersProfileEditPasword from 'components/templates/teachers/profile/EditPassword'
import * as React from 'react'

export default () => {
  return <TeachersProfileEditPasword />
}
