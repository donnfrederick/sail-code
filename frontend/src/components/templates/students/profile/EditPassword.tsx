import Footer from 'components/molecules/students/Footer'
import EditPassword from 'components/organisms/students/edit_password'
import Header from 'components/organisms/students/header'
import * as React from 'react'
import styled from 'styled-components'

export default () => {
  return (
    <Container>
      <EditPassword />
      <Header text="Change Password" hasBackButton={true} />
      <Footer />
    </Container>
  )
}

const Container = styled.div`
  position: relative;
  width: 100%;
  min-height: calc(var(--vh, 1vh) * 100);
  box-sizing: border-box;
  padding: 360px 0;
`
