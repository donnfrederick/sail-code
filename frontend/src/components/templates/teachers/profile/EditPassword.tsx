import EditPageTitle from 'components/atoms/teachers/EditPageTitle'
import Header from 'components/molecules/teachers/Header'
import EditPageButtons from 'components/organisms/teachers/edit_page_buttons'
import EditPassword from 'components/organisms/teachers/edit_password'
import * as React from 'react'
import styled from 'styled-components'

export default () => {
  return (
    <Container>
      <EditPageTitle
        text={'新しいパスワードを\n入力してください'}
        marginBottom={84}
      />
      <EditPassword />
      <EditPageButtons type="password" />
      <Header hasSupport={true} text="パスワードを変更" backToHome={true} />
    </Container>
  )
}

const Container = styled.div`
  position: relative;
  width: 100%;
  min-height: calc(var(--vh, 1vh) * 100);
  box-sizing: border-box;
  padding: 192px 0 56px;
  text-align: center;
`
