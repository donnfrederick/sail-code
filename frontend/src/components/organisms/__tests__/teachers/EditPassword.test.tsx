import Router from 'components/organisms/Router'
import EditPassword from 'components/organisms/teachers/edit_password'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <EditPassword />', () => {
  const tree = renderer
    .create(
      <Router>
        <EditPassword />
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
