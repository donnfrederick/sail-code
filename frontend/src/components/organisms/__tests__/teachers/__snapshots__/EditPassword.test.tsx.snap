// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <EditPassword /> 1`] = `
<div
  className="sc-EHOje cpaeQV"
>
  <div
    className="sc-bZQynM hACQYa"
    data-error={false}
  >
    現在のパスワード
  </div>
  <input
    className="sc-gzVnrw kLVULG"
    data-error={false}
    maxLength={72}
    onInput={[Function]}
    pattern="^[0-9A-Za-z]+$"
    placeholder="半角英数8字以上"
    type="password"
  />
  <div
    className="sc-bZQynM hACQYa"
    data-error={false}
  >
    新しいパスワード
  </div>
  <input
    className="sc-gzVnrw kLVULG"
    data-error={false}
    maxLength={72}
    onInput={[Function]}
    pattern="^[0-9A-Za-z]+$"
    placeholder="半角英数8字以上"
    type="password"
  />
</div>
`;
